### PR TITLE
Add test cases from partial_evaluator

### PR DESCRIPTION
**Pull request by ayazhafiz**
This is a copy of partial_evaluator tests for output and s-expression
cases.

### CODE REVIEW

File: slide/src/test/ui/distribute_negation.slide [hunk 1]
```diff
@@ -0,0 +1,15 @@
...
+~~~stdout
+
+~~~stderr
+error: Found argument '-(' which wasn't expected, or isn't valid in this context
+
+USAGE:
+    slide [FLAGS] [OPTIONS] <program>
+
+For more information try --help
+~~~stderr
```
Comment (ayazhafiz) [inline slide/src/test/ui/distribute_negation.slide:15]: Ahh this is a bug

File: slide/src/test/ui/negate.slide [hunk 2]
```diff
@@ -0,0 +1,15 @@
...
+~~~stdout
+
+~~~stderr
+error: Found argument '-1' which wasn't expected, or isn't valid in this context
+
+USAGE:
+    slide [FLAGS] [OPTIONS] <program>
+
+For more information try --help
+~~~stderr
```
Comment (ayazhafiz) [inline slide/src/test/ui/negate.slide:15]: same here

File: slide/src/test/ui/distribute_negation.slide (hunk 1 above)
Comment (ayazhafiz) [reply slide/src/test/ui/distribute_negation.slide:15]: Going forward, I'm planning to add two more options:
- `~~~exitcode`
- `@TODO: ...` as a directive, with an option to fail on TODO tests

File: slide/src/test/ui/output_form/s_expression/unwrap_parens_var.slide [hunk 3]
```diff
@@ -0,0 +1,19 @@
...
+--parse-only
+-o s-expression
+!!!args
+
+===in
+(a)
+===in
+
+~~~stdout
+(a)
```
Comment (timothycobrien) [inline slide/src/test/ui/output_form/s_expression/unwrap_parens_var.slide:11]: Shouldn't this just be `a` or am I missing something?

Comment (ayazhafiz) [reply slide/src/test/ui/output_form/s_expression/unwrap_parens_var.slide:11]: We stop at the parsing phase, see `--parse-only` in the args above

**Pull request merged**